src/general/OptionsMenu.cs

```diff
@@ -461,6 +468,25 @@ private void CompareSettings()
         }
     }
 
+    private void LoadLanguages(OptionButton optionButton)
+    {
+        Array locals = TranslationServer.GetLoadedLocales();
+
+        foreach (string local in locals)
+        {
+            optionButton.AddItem(local);
+        }
+
+        if (optionButton.Items.Count == 0)
+            return;
+
+        // The option button seems to have 5 items / options.
+        // This means that the second option has the index 5 in the items list, but Selected need the id 1 to works.
+        // This is strange.
+        optionButton.Selected = optionButton.Items.IndexOf(Settings.Instance.SelectedLanguage) / 5;
```

Comment:
Hardcoding 5 here is not very good, as I expect that to change when we add more translations, another approach is needed.

Reply:
I don't think it will change. Each new option in the options button will always add 5 items. So in theory, this should work with any amounts of localization we add.

eg. 3 localizations, the index of the third string in the items list will be 10. But when setting selected, we want 10/5 = 2

Here is an example of those items from the MSAAResolution option button :
items = [ "Disabled", null, false, 0, null, 
               "2x", null, false, 1, null, 
               "4x", null, false, 2, null, 
               "8x", null, false, 3, null, 
               "16x", null, false, 4, null ]

There may be another way to do this though, I'm not yet familiar with how Godot works.

Reply:
I haven't looked into this myself, but it would be nice to have a simpler solution for setting the selected item.
How do the other buttons do it? They have 0 based indexes that go 0-4 and that just works fine? Why doesn't that work here?
Time to summon our Godot GUI specialist @Kasterisk ...

Reply:
Very interesting, so I found a related godot issue https://github.com/godotengine/godot/issues/40258. It turns out `Items` property is an array containing data for all the items itself (hence 5 elements each, the first one is the item's text), while `GetItemCount()` actually returns the actual count of items on the option button. So maybe a more "correct" way to select item by text is something like this?
```
for (var index = 0; index < optionButton.GetItemCount(); index++)
        {
            if (optionButton.GetItemText(index) == Settings.Instance.SelectedLanguage)
            {
                optionButton.Selected = index;
            }
        }
```
Doesn't seem more simpler tbh haha :upside_down_face:

Reply:
Not a huge fan of adding a loop just for that, but it's your call @hhyyrylainen

Reply:
Can't we just store the languages in a List and look up the index from that whenever we need?
Because this code works: https://github.com/Revolutionary-Games/Thrive/blob/1bddded590c69a6f96b4353d48c5f2f5bf418404/src/general/OptionsMenu.cs#L198
https://github.com/Revolutionary-Games/Thrive/blob/1bddded590c69a6f96b4353d48c5f2f5bf418404/src/general/OptionsMenu.cs#L343

Reply:
That would mean every time someone wants to add a new language, they'll have to add it to the List in a cs file, while right now they just need to create the .po file, and add it into the language setting in Godot.

I don't think making adding a new language more complicated is worth it for a single line of code, since I guess we'll want to make translation work as easy as possible for people with no dev background.

And btw, Viewport.MSAA being an Enum, as long as the options are in the same order in the option button you should be able to use for example :

From res to index: return (int) res;
From index to res : return (Viewport.MSAA) index

Reply:
I meant that the C# List would be automatically generated when the options menu object is loaded.
So there would be no need to change the C# source code when new translations are added.

> From res to index: return (int) res;
From index to res : return (Viewport.MSAA) index

I think that could be a pretty good change to do in a small PR, to reduce the amount of needed code for those options.

Reply:
Then I misunderstood what you meant. This indeed allows us to improve the code.

Just pushed a commit for that.